fabric/bin/peer.sh

```diff
@@ -254,6 +254,12 @@ handle_channel_join() {
     # - call real peer so channel is joined
     try $RUN ${FABRIC_BIN_DIR}/peer "${ARGS_EXEC[@]}"
 
+    # - ask tlcc to join channel
+    #   IMPORTANT: right now a join is _not_ persistant, so on restart of peer,
+    #   it will re-join old channels but tlcc will not!
+    say "Attaching TLCC to channel '${CHAN_ID}' ..."
+    try $RUN ${FABRIC_BIN_DIR}/peer chaincode query -n tlcc -c '{"Args": ["JOIN_CHANNEL"]}' -C ${CHAN_ID}
```

Comment:
query causes a join?

Reply:
curious, why did you change order of ercc and tlcc add/init?

Reply:
tlcc should first join the channel and thereby register for new blocks/transactions. So this must happen as first action of a FPC Peer when joining a channel; otherwise tlcc will not see any transactions related to ercc lifecycle/installation. Indeed, those lifecycle transactions are currently not really processed within tlcc.

Reply:
Does this make sense?

Reply:
Hmm, i thought we always read from the genesis block, so it wouldn't matter unless ercc requires tlcc?  If what you say is indeed true, it would also mean that what we claim that we currently support peer restart would be wrong?!

Reply:
let me double check -

Reply:
Still confused about this, maybe a design issue to resolve later. I would think we need to write the tlcc config to the chain to trust it.

Not confused about the order. Agree that this means peer restart does not work unless the txn is re-run.

Reply:
aha! @MHBauer I think you refer to the MRENCLAVE of TLCC itself. How do we get this into TLCC; this critical feature is not implemented yet.

Reply:
> TLCC crashes because it tried to read MRENCLAVE from a readset when validating a ercc transaction; however, with PR #330 we removed init that writes MRENCLAVE to the ledger. I propose to move it for now to the _setup as we will fetch MRENCLAVE from the chaincode definition.

From what we talked in phoenix, the plan was to bake it into ercc code.  But then with the new protocols the bootstrapping will anyway also require changes in ecc and elsewhere.  In any case, you also mean `ecc`, not `ercc`.  In which case moving it to `__setup` seems a good idea ...